CMakeLists.txt

```diff
@@ -146,12 +146,13 @@ endif(BUILD_PYTHON_INTERFACE)
 macro(create_demo source)
 
   set(demo_name ${PROJECT_NAME}_${source})
+  set(DEMO_ROOT "${PROJECT_SOURCE_DIR}/demos")
   add_executable(${demo_name} demos/${source}.cpp)
   target_link_libraries(${demo_name} ${PROJECT_NAME})
   target_compile_definitions(
     ${demo_name}
-    PUBLIC CONFIG_SOLO12_YAML="${PROJECT_SOURCE_DIR}/demos/config_solo12.yaml"
-    PUBLIC CONFIG_TESTBENCH_YAML="${PROJECT_SOURCE_DIR}/demos/config_testbench.yaml")
+    PUBLIC CONFIG_SOLO12_YAML="${DEMO_ROOT}/config_solo12.yaml"
+    PUBLIC CONFIG_TESTBENCH_YAML="${DEMO_ROOT}/config_testbench.yaml")
```

Comment:
passing paths to the demo binaries forces us to have to files available at runtime, which I did manage to work around by patching `set(DEMO ROOT` but its pretty ugly and requires dumping extra stuff into $out, I'd bake the demo yaml files' *contents* into the binary instead, what do you think? I started working on it but i'm not sure if it's worth the extra code

Reply:
![image](https://github.com/user-attachments/assets/78498b0c-044d-4418-a8f7-be1f877f4f7d)

I can't really pass a file's contents to the compiler anyways... people use xxd to generate a .c file but... this means another compile-time dep, .... honestly im not sure if its worth it